### PR TITLE
Update associated test target comparison for added flexibility.

### PR DESCRIPTION
**Pull request by jakeatoms**
Resolves https://github.com/tuist/tuist/issues/2796
Request for comments document (if applies):

### Short description 📝

This PR uses a regular expression in place of hardcoded options for finding potential associated test targets for a target scheme.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

File: Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift [hunk 1]
```diff
@@ -139,9 +139,7 @@ public final class AutogeneratedSchemesProjectMapper: ProjectMapping {
 
     private func isAssociatedTestTarget(for target: Target, potentialTestTarget: Target) -> Bool {
         let isTestsBundle = potentialTestTarget.product.testsBundle
-        let isNameMatching = potentialTestTarget.name == "\(target.name)Tests" ||
-            potentialTestTarget.name == "\(target.name)UITests" ||
-            potentialTestTarget.name == "\(target.name)IntegrationTests"
+        let isNameMatching = potentialTestTarget.name.range(of: #"\#(target.name)\w*Tests"#, options: .regularExpression) != nil
```
Comment (jakeatoms) [inline Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift:142]: I am far from a regex expert, so please suggest improvements if you have any!

Comment (kalkwarf) [reply Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift:142]: That looks correct to me.

Comment (kwridan) [reply Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift:142]: Perhaps a regex isn't needed here - a slightly faster alternative that achieves the same result is

```swift
potentialTestTarget.name.hasPrefix(target.name) && potentialTestTarget.name.hasSuffix("Tests")
```

**Review by kwridan — COMMENTED** [review]
Thanks for submitting this @jakeatoms 

I guess no matter which approach we follow there will still be uncovered cases where we either pick up too many or too few targets - but its ok those are just defaults, to fine tune the end results users can specify custom schemes 🙂

File: Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift (hunk 1 above)
Comment (jakeatoms) [reply Sources/TuistGenerator/Mappers/AutogeneratedSchemesProjectMapper.swift:142]: I like that. It's more immediately clear. I'll update and fix formatting.

**Comment by jakeatoms** [comment]
> Thanks for submitting this @jakeatoms
> 
> I guess no matter which approach we follow there will still be uncovered cases where we either pick up too many or too few targets - but its ok those are just defaults, to fine tune the end results users can specify custom schemes 🙂

Agreed. This approach is a little more flexible and maintains the old behavior, which we rely on for 30 or so frameworks in our workspace. We're eventually going to move to strictly custom schemes, but we need some other architecture in place before that becomes feasible.

Thanks!

**Review by kwridan — APPROVED** [review]

**Pull request merged**